app/parse.py

```diff
@@ -15,8 +27,57 @@ class Product:
     num_of_reviews: int
 
 
+def get_products_from_page(driver: WebDriver, url: str) -> list[Product]:
+    driver.get(url)
+    time.sleep(1)
+    driver.refresh()
+    while True:
+        try:
+            button = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        except NoSuchElementException:
+            break
+        if button.get_attribute("style") == "display: none;":
```

Comment:
You are clicking the "scroll more" button with a while loop. However, the exit condition may not always work due to the nature of page loading. Ensure that the logic for detecting the end of content (button.get_attribute("style") == "display: none;") works reliably.